### ISSUE TITLE
`print` order broken while piping

### ISSUE THREAD

**Issue by andry81** [xonsh/xonsh#3202]
## xonfig

<details>

```
+------------------+-------------------------------------------------------+
| xonsh            | 0.9.7                                                 |
| Git SHA          | 626b94db                                              |
| Commit Date      | Jun 28 17:20:28 2019                                  |
| Python           | 3.7.3                                                 |
| PLY              | 3.11                                                  |
| have readline    | False                                                 |
| prompt toolkit   | 2.0.9                                                 |
| shell type       | prompt_toolkit2                                       |
| pygments         | None                                                  |
| on posix         | False                                                 |
| on linux         | False                                                 |
| on darwin        | False                                                 |
| on windows       | True                                                  |
| on cygwin        | <xonsh.lazyasd.LazyBool object at 0x0000000002ADA630> |
| on msys2         | <xonsh.lazyasd.LazyBool object at 0x0000000002ADA518> |
| is superuser     | False                                                 |
| default encoding | utf-8                                                 |
| xonsh encoding   | utf-8                                                 |
| encoding errors  | surrogateescape                                       |
+------------------+-------------------------------------------------------+
```

</details>

## Expected Behavior
Correct print output order, for example, as in the Bash shell or any other shell.

## Current Behavior
Print output is reordered.

## Steps to Reproduce
**1.bat**
```
@echo off

setlocal

(
  python -m xonsh 1.xsh
) 2>&1 | more
```

**1.xsh**
```
import sys, os, inspect

SOURCE_FILE = os.path.abspath(inspect.getsourcefile(lambda:0)).replace('\\','/')
#SOURCE_DIR = os.path.dirname(SOURCE_FILE)
$SOURCE_DIR = os.path.dirname(SOURCE_FILE)

print("1 ->", SOURCE_FILE)

#evalx("source '{0}'".format(os.path.join(SOURCE_DIR, '2.xsh').replace('\\', '/')))
source $SOURCE_DIR/2.xsh
```

**2.xsh**
```
SOURCE_FILE = os.path.abspath(inspect.getsourcefile(lambda:0)).replace('\\','/')

print("2 ->", SOURCE_FILE)
```

**Output**:
```
2 ->1 -> c:/Python/x64/37/1.xsh
 c:/Python/x64/37/2.xsh
```

I know that I can use `echo` instead. But it tries to create an external process: `cmd.exe /c echo ...`, which is not what I expect or want. Secondly, it anyway get hangs.

All this is pretty sad, because I can not create the log redirection internally: https://github.com/xonsh/xonsh/issues/3198
nor externally, because the output order becomes mixed.

Which means that the both redirection strategy are broken and so, get draw the xonsh completely unusable in this field.

**Comment by astronouth7303** [comment]
xonsh struggles with stdout vs stderr ordering because of the way POSIX and xonsh works, and fixing would require sacrificing in other areas.

**Comment by andry81** [comment]
Re: [xonsh/xonsh] `print` order broken while piping (#3202)







xonsh struggles with stdout vs stderr ordering because of the way POSIX and xonsh works, and fixing would require sacrificing in other areas.


I am not quite sure why this has to be a sacrifice :)
May be you have to think how to change the design to fix that, isn't?

**Comment by scopatz** [comment]
This is pretty strange, the behavior you are seeing.  I see what you would expect on linux:

```
$ xonsh 1.xsh
1 -> /home/scopatz/temp/1.xsh
2 -> /home/scopatz/temp/2.xsh
```

**Comment by scopatz** [comment]
Also, which issue are you referencing? In this case, stdout should always be ordered properly.  Only stdout & stderr have no guarantee on when they are printed to the same screen.

**Comment by andry81** [comment]
I didn't test stdout & stderr. Only what in the example.
I have tried to just replace the existing code with the simple redirection and found that the stdout is mixed.

**Comment by scopatz** [comment]
Can anyone reproduce this on windows? @melund? This seems like a bug.

**Comment by andry81** [comment]
seems fixed in `0.9.13.dev1`
